types/svelte-range-slider-pips/index.d.ts

```diff
@@ -0,0 +1,165 @@
+    /**
+     * Make the slider render vertically (lower value on bottom)
+     */
+    vertical?: boolean;
+    /**
+     * Set true to add a floating label above focussed handles
+     */
+    float?: boolean;
+    /**
+     * Reverse the orientation of min/max
+     */
+    reversed?: boolean;
+    /**
+     * Whether hover styles are enabled for both handles and pips/values
+     */
+    hoverable?: boolean;
+    /**
+     * Determine if the slider is disabled, or enabled
+     * (only disables interactions, and events)
+     */
+    disabled?: boolean;
+
+    // range pips / values props
+    /**
+     * Whether to show pips/notches on the slider
+     */
+    pips?: boolean;
+    /**
+     * Every nth step to show a pip for.
+     * This has multiple defaults depending on values property
+     */
+    pipstep?: number;
+    /**
+     * Whether to show a pip or label for all values.
+     * Same as combining first, last and rest. Use all='label' to show a label value
+     */
+    all?: boolean | string;
+    /**
+     * Whether to show a pip or label for the first value on slider.
+     * Use first='label' to show a label value
+     */
+    first?: boolean | string;
+    /**
+     * Whether to show a pip or label for the last value on slider.
+     * Use last='label' to show a label value
+     */
+    last?: boolean | string;
+    /**
+     * Whether to show a pip or label for all other values.
+     * Use rest='label' to show a label value
+     */
+    rest?: boolean | string;
+
+    // formatting props
+    /**
+     * Give the slider a unique ID for use in styling
+     */
+    id?: string;
+    /**
+     * A string to prefix to all displayed values
+     */
+    prefix?: string;
+    /**
+     * A string to suffix to all displayed values
+     */
+    suffix?: string;
+    /**
+     * A function to re-format values before they are displayed (v = value, i = pip index, p = percent)
+     */
+    formatter?: Formater;
+    /**
+     * A function to re-format values on the handle/float before they are displayed. Defaults to the same function given to the formatter property (v = value, i = handle index, p = percent)
+     */
+    handleFormatter?: Formater;
+
+    // stylistic props
+    precision?: number;
+    /**
+     * Svelte spring physics object to change the behaviour of the handle when moving
+     */
+    springValues?: SpringValues;
+}
+
+interface RangeSliderEvents {
+    /**
+     * Event fired when the user begins interaction with the slider
+     */
+    start: CustomEvent<OnStart>;
+    /**
+     * Event fired when the user changes the value; returns the previous value, also
+     */
+    change: CustomEvent<OnChange>;
+    /**
+     * Event fired when the user stops interacting with slider; returns the beginning value, also
+     */
+    stop: CustomEvent<OnStop>;
+}
+
+type RangeSliderSlots = never;
+
+export default class RangeSlider extends SvelteComponentTyped<RangeSliderProps, RangeSliderEvents, RangeSliderSlots> {}
```

Comment:
To switch to export =, you'll need to `declare class RangeSlider` separately, then `export = RangeSlider`.

And instead of exporting the types below, you'll need to put them inside `namespace RangeSlider` and refer to them as `RangeSlider.OnStart` (for example) in the tests.